compose/ui/ui/src/commonMain/kotlin/androidx/compose/ui/platform/CompositionLocals.kt

```diff
@@ -188,31 +188,40 @@ internal val LocalPointerIconService = staticCompositionLocalOf<PointerIconServi
 @Composable
 internal fun ProvideCommonCompositionLocals(
     owner: Owner,
+    isMainOwner: Boolean,
     uriHandler: UriHandler,
     content: @Composable () -> Unit
 ) {
     CompositionLocalProvider(
-        LocalAccessibilityManager provides owner.accessibilityManager,
-        LocalAutofill provides owner.autofill,
-        LocalAutofillTree provides owner.autofillTree,
-        LocalClipboardManager provides owner.clipboardManager,
-        LocalDensity provides owner.density,
         LocalFocusManager provides owner.focusOwner,
```

Comment:
I think ideally, the algorithm for deciding whether to override each local is:
1. If the existing local is not provided (we're setting up the main owner) - provide it.
2. Otherwise, if the existing local is user-provided - do not override it.
3. Otherwise, if the existing local uses the owner - override it.
4. Otherwise, don't override it.

But (2) is problematic:
- How do we know whether the existing local is user-provided.
- It could cause unexpected issues. For example if the user-provided local is just wrapping our local and changing/adding some behavior.

So maybe we should forego (2) and just do:
1. If the existing local is not provided (we're setting up the main owner) - provide it.
2. Otherwise, if the local uses the owner - override it.
3. Otherwise, don't override it.

The question then is how to ask the owner whether the local uses the owner. We could
1. Have the owner return a `Pair<T, Boolean>` for each local value, e.g.
 ```val clipboardManager = PlatformClipboardManager() to false```
2. Replace the `val`s with `fun`s, e.g. 
  ```fun clipboardManager(current: ClipboardManager?) = current ?: PlatformClipboardManager()```
3. Have the owner provide the locals itself, i.e. just move `ProvideCommonCompositionLocals` into `Owner`. By the way, most of the locals exposed by the owner are only ever used in `ProvideCommonCompositionLocals`, so we could remove them from the API altogether.

Reply:
As the changes are in common code should not we discuss them with Google as well? Now again we provoke different behavior with Android target.

Reply:
@pjBooms That's an excellent point. I'll talk to the Google folk.